### ISSUE TITLE
Better errors for unrecognised command line options

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript#1838]
At the moment, if you pass a command line option that `psc` (or any of the other tools) do not recognise, it won't tell you which option wasn't recognised. This can be particularly irritating when you're using tools that invoke `psc`, because it's not always clear what the actual `psc` invokation was. Can `psc` and the other compiler command line tools say which options were unrecognised, if any? 

Currently, it looks like this:
```
$ psc --lol
Usage: psc [FILE] [-f|--ffi ARG] [-o|--output ARG] [--no-tco] [--no-magic-do]
           [--no-opts] [-v|--verbose-errors] [-c|--comments]
           [-r|--require-path ARG] [-p|--no-prefix]
```

Ideally, I think it would look like this:
```
$ psc --lol
Usage: psc [FILE] [-f|--ffi ARG] [-o|--output ARG] [--no-tco] [--no-magic-do]
           [--no-opts] [-v|--verbose-errors] [-c|--comments]
           [-r|--require-path ARG] [-p|--no-prefix]
psc: Unrecognised flag: --lol
```

One example of how this can cause pain is https://github.com/bodil/pulp/pull/163#issuecomment-172675944

**Comment by garyb** [comment]
:+1:

**Comment by stevejb71** [comment]
I was looking for an easy issue to work on. This one seems to work already in the from-source build:

psc --lol
Invalid option `--lol'

Usage: psc [FILE] [-f|--ffi ARG] [-o|--output ARG] [--no-tco] [--no-magic-do]
           [--no-opts] [-v|--verbose-errors] [-c|--comments]
           [-r|--require-path ARG] [--source-maps] [-p|--no-prefix]
           [--json-errors]

**Comment by hdgarrood** [comment]
@stevejb71 Ah, nice! Same for me on `master`. Seems this is implemented in `optparse-applicative` as of 0.12.1.0: http://hackage.haskell.org/package/optparse-applicative-0.12.1.0/changelog.

Perhaps we should require optparse-applicative >= 0.12.1.0 in purescript.cabal?

**Comment by paf31** [comment]
A version bump sounds good.
